frontend/lib/resolution/resolution-queries.cpp

```diff
@@ -3229,7 +3241,10 @@ static bool shouldAttemptImplicitReceiver(const CallInfo& ci,
                                           QualifiedType implicitReceiver) {
   return !ci.isMethodCall() &&
          !ci.isOpCall() &&
-         implicitReceiver.type() != nullptr;
+         implicitReceiver.type() != nullptr &&
+         // Assuming ci.name().isEmpty()==true implies a primitive call.
+         // TODO: Add some kind of 'isPrimitive()' to CallInfo
+         !ci.name().isEmpty();
```

Comment:
The name can also be empty if it is a call to a first-class-function. In which case the base expression is a non-param function value.